producer.go

```diff
@@ -99,7 +98,13 @@ func (p *Producer) consumeDeliveries() {
 			}
 		} else {
 			p.log.Printf("Error consuming delivery event: Unknown event type (%s)", ev)
-			atomic.AddUint64(&stats.producerErr, 1)
+
+			// Ignore 'connection reset by peer' errors since they manifest because
+			// of the connection reaper.
+			// See https://github.com/edenhill/librdkafka/wiki/FAQ#why-am-i-seeing-receive-failed-disconnected.
+			if !strings.Contains(ev.String(), "Connection reset by peer") {
```

Comment:
What's the value returned by `ev.String()`? Maybe we could instead use instead `strings.HasPrefix` to be more explicit and err on the safe side (i.e. reducing the possibility of ignoring something we shouldn't).

Reply:
From the logs we can tell that exact value of ev.String() is `broker:9092/4: Receive failed: Connection reset by peer (after 1199998ms in state UP)`.

Reply:
OK nvm then. Ignore my comment.